File: aire-api/src/main/java/io/aire/designer/Image.java
package io.aire.designer;

public class Image {

  private int width;
  private int height;
  private String source;
  private ImageType type;

  public Image(ImageType type, String source, int width, int height) {
    this.type = type;
    this.width = width;
    this.height = height;
    this.source = source;
  }

  public int getWidth() {
    return width;
  }

  public int getHeight() {
    return height;
  }

  public String getSource() {
    return source;
  }

  public ImageType getType() {
    return type;
  }
}

File: aire-core/src/main/frontend/application/aire/designer/core/image.ts
/**
 * type of image (handlers will know how to deal with this)
 */
export interface ImageType {
  getKey() : string;
}


/**
 * actual image class
 */
export class Image {
  width : number;
  height : number;
  type : ImageType;
  source : string;
}


/**
 * internal members
 */
export class BaseImageType implements ImageType {
  constructor(private readonly key : string) {

  }

  getKey() : string {
    return this.key;
  }
}


export class ImageTypes {
  public static readonly Palette : ImageType = new BaseImageType("palette");
  public static readonly Element : ImageType = new BaseImageType("element");
}
